sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/_client.py

```diff
@@ -165,6 +184,10 @@ def decrypt(self, algorithm, ciphertext, **kwargs):
             :dedent: 8
         """
         self._initialize(**kwargs)
+        iv = kwargs.pop("iv", None) if ("CBC" in algorithm or "GCM" in algorithm) else None
+        tag = kwargs.pop("authentication_tag", None) if "GCM" in algorithm else None
+        aad = kwargs.pop("additional_authenticated_data", None) if "GCM" in algorithm else None
```

Comment:
If I call `decrypt(EncryptionAlgorithm.rsa_oaep, b".", iv=b".", tag=b".", aad=b".")`, my superfluous arguments stay in `kwargs` and may be passed along to the generated client.

But it's a good idea to reject useless arguments. Silently ignoring them is risky though, because a caller could reasonably infer they had an effect. Looks like it's easy with the current set of algorithms to identify invalid arguments--what do you think of raising `ValueError` when we get them?

Reply:
I think a `ValueError` sounds like a good idea -- it would be a good opportunity to raise an explanatory message as well